src/buildSummary.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 The MathWorks, Inc.
+import * as core from "@actions/core";
+import { join } from 'path';
+import { readFile } from 'fs';
+import { promisify } from 'util';
+
+// Promisify the readFile function to use it with async/await
+const readFileAsync = promisify(readFile);
+
+export interface Task {
+  name: string;
+  description: string;
+  failed: boolean;
+  skipped: boolean;
+  duration: string;
+}
+
+export interface TaskList {
+  taskDetails: Task[];
+}
+
+export async function readJsonFile(filePath: string): Promise<TaskList> {
+  try {
+    const data = await readFileAsync(filePath, { encoding: 'utf8' });
+    return JSON.parse(data);
+  } catch (error) {
+    console.error('Error reading the JSON file:', error);
+    throw error;
+  }
+}
+
+export function getBuildSummaryTable(tasks: TaskList): string[][] {
+  const header: string[] = ['MATLAB Build Task', 'Status', 'Description', 'Duration (HH:MM:SS)'];
+  let taskSummaryTableRows: string[][] = [header];
+
+  tasks.taskDetails.forEach((task, index) => {
+    let taskDetails: string[] = [];
+    taskDetails.push(task.name);
+    if (task.failed) {
+      taskDetails.push('🔴 FAILED');
+    } else if (task.skipped) {
+      taskDetails.push('🔵 SKIPPED');
+    } else {
+      taskDetails.push('🟢 SUCCESS');
+    }
+    taskDetails.push(task.description);
+    taskDetails.push(task.duration);
+
+    taskSummaryTableRows.push(taskDetails);
+  });
+
+  return taskSummaryTableRows;
+}
+
+export function writeSummary(taskSummaryTableRows: string[][]) {
+    core.summary
+      .addTable(taskSummaryTableRows)
+      .write();
+}
+
+export async function processAndDisplayBuildSummary() {
+  const runId = process.env.GITHUB_RUN_ID;
+
+  if (!runId) {
+    console.error('GITHUB_RUN_ID environment variable is not set. Unable to locate the build summary file.');
+    return;
+  }
+
+  try {
+    const runnerTemp = process.env.RUNNER_TEMP;
+    const filePath = join(runnerTemp as string, `buildSummary_${runId as string}.json`);
+    const data = await readJsonFile(filePath);
+    const taskSummaryTableRows = getBuildSummaryTable(data);
+    writeSummary(taskSummaryTableRows);
+  } catch (error) {
+    console.error('An error occurred while reading the build summary file or adding the build summary table:', error);
```

Comment:
This error seems a bit generic and probably not particularly hepful to users. 

Currently we're already logging an error message if reading the build summary file fails (line 27) and then rethrowing.

Could we instead just not rethrow and additionally add a similar error message scoped just for writing to the summary failing in `writeSummary`?

Reply:
Something like
```typescript
export function writeSummary(taskSummaryTableRows: string[][]) {
    try {
        core.summary
            .addTable(taskSummaryTableRows)
            .write();
    } catch (e) {
        console.error('An error occurred while adding the build results table to the summary');
    }
}
```

And then this code could just be
```typescript
const runnerTemp = process.env.RUNNER_TEMP;
const filePath = join(runnerTemp as string, `buildSummary_${runId as string}.json`);
const data = await readJsonFile(filePath);
const taskSummaryTableRows = getBuildSummaryTable(data);
writeSummary(taskSummaryTableRows);
```

with no `try`/`catch` logic.

Reply:
Please consider the the suggestion here 
https://github.com/matlab-actions/run-build/pull/33#discussion_r1655257446